make/compiler/Makefile.mpi-gnu

```diff
@@ -19,5 +19,13 @@
 
 include $(CHPL_MAKE_HOME)/make/compiler/Makefile.gnu
 
-CC=mpicc
-CXX=mpicxx
+ifeq (, $(MPI_CC))
+MPI_CC=mpicc
+endif
+
+ifeq (, $(MPI_CXX))
+MPI_CXX=mpicxx
+endif
+
+CC=$(MPI_CC)
+CXX=$(MPI_CXX)
```

Comment:
My first thought was "shouldn't we create an mpi-intel equivalent to this and set the relevant MPI_ variables there siilarly?  But then I couldn't find places where this Makefile is still used and wonder if it dates back to our CHPL_COMM=mpi days (in which case it should probably be retired).

Reply:
Ah good point -- this is still used as part of the MPI module that Nikhil added (and we test this config nightly)

Reply:
So we may want to add a mpi-intel one as well, but unless you object I'll probably leave this change as since I don't think it hurts anything

Reply:
> this is still used as part of the MPI module that Nikhil added

Oh, I see.  So the reason I wasn't finding it by grepping is that the user has to set up for it explicitly.  Makes sense (and I didn't think to git grep for "mpi-gnu" which would've gotten me there...